File: src/test/java/com/flexpoker/table/command/aggregate/pot/PotTestUtils.java
package com.flexpoker.table.command.aggregate.pot;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

import com.flexpoker.table.command.HandRanking;
import com.flexpoker.table.command.CardRank;
import com.flexpoker.table.command.aggregate.HandEvaluation;

public class PotTestUtils {

    static PotHandler createBasicPotHandler(UUID player1, UUID player2) {
        var handEvaluation1 = new HandEvaluation();
        handEvaluation1.setPlayerId(player1);
        handEvaluation1.setHandRanking(HandRanking.FLUSH);
        handEvaluation1.setPrimaryCardRank(CardRank.EIGHT);
        handEvaluation1.setFirstKicker(CardRank.SEVEN);
        handEvaluation1.setSecondKicker(CardRank.FOUR);
        handEvaluation1.setThirdKicker(CardRank.THREE);
        handEvaluation1.setFourthKicker(CardRank.TWO);
        var handEvaluation2 = new HandEvaluation();
        handEvaluation2.setPlayerId(player2);
        handEvaluation2.setHandRanking(HandRanking.STRAIGHT);
        handEvaluation2.setPrimaryCardRank(CardRank.KING);

        var winningHands = new ArrayList<HandEvaluation>();
        winningHands.add(handEvaluation1);
        winningHands.add(handEvaluation2);

        return new PotHandler(UUID.randomUUID(), UUID.randomUUID(),
                UUID.randomUUID(), winningHands);
    }

    static Set<UUID> createSetOfPlayers(UUID... players) {
        return new HashSet<>(Arrays.asList(players));
    }

}

File: src/test/kotlin/com/flexpoker/table/command/aggregate/pot/PotTestUtils.kt
package com.flexpoker.table.command.aggregate.pot

import com.flexpoker.table.command.CardRank
import com.flexpoker.table.command.HandRanking
import com.flexpoker.table.command.aggregate.HandEvaluation
import java.util.UUID

fun createPotHand(playerId: UUID): HandEvaluation {
    return HandEvaluation(
        playerId = playerId,
        handRanking = HandRanking.FLUSH,
        primaryCardRank = CardRank.EIGHT,
        firstKicker = CardRank.SEVEN,
        secondKicker = CardRank.FOUR,
        thirdKicker = CardRank.THREE,
        fourthKicker = CardRank.TWO,
    )
}

fun createPotHands(player1: UUID, player2: UUID): List<HandEvaluation> {
    val handEvaluation1 = createPotHand(player1)
    val handEvaluation2 = HandEvaluation(
        playerId = player2,
        handRanking = HandRanking.STRAIGHT,
        primaryCardRank = CardRank.KING,
    )
    return listOf(handEvaluation1, handEvaluation2)
}